src/applications/lgy/coe/form/tests/coe-keyboard-only.cypress.spec.js

```diff
@@ -0,0 +1,194 @@
+          );
+          cy.typeInDate(
+            `root_periodsOfService_${index}_dateRange_to`,
+            data.periodsOfService[index].dateRange.to,
+          );
+          if (index + 1 < toursLength) {
+            cy.tabToElement('.va-growable-add-btn');
+            cy.realPress('Space');
+          }
+        });
+      }
+    });
+    cy.tabToContinueForm();
+
+    // Loan history
+    cy.url().should('include', chapters.loansChapter.pages.loanScreener.path);
+    cy.tabToElement('[name="root_vaLoanIndicator"]');
+    cy.chooseRadio(data.vaLoanIndicator ? 'Y' : 'N');
+    cy.tabToContinueForm();
+
+    // Loan intent
+    cy.url().should('include', chapters.loansChapter.pages.loanIntent.path);
+    cy.tabToElement('[name="root_intent"]');
+    cy.chooseRadio(data.intent);
+    cy.tabToContinueForm();
+
+    // Loan history
+    // Only adding one loan because tabbing takes a lot of time
+    cy.url().should('include', chapters.loansChapter.pages.loanHistory.path);
+    if (data.relevantPriorLoans.length) {
+      const index = 0;
+      const root = `root_relevantPriorLoans_${index}_`;
+      const firstLoan = data.relevantPriorLoans?.[index];
+
+      const from = firstLoan.dateRange.from
+        .split('-')
+        .map(v => parseInt(v, 10).toString());
+      cy.tabToElement(`#${root}dateRange_fromMonth`);
+      cy.chooseSelectOptionUsingValue(from[1]);
+      cy.tabToElement(`input[name="${root}dateRange_fromYear"]`);
+      cy.typeInFocused(from[0]);
+
+      const to = firstLoan.dateRange.to
+        .split('-')
+        .map(v => parseInt(v, 10).toString());
+      cy.tabToElement(`#${root}dateRange_toMonth`);
+      cy.chooseSelectOptionUsingValue(to[1]);
+      cy.tabToElement(`input[name="${root}dateRange_toYear"]`);
+      cy.typeInFocused(to[0]);
+
+      cy.typeInIfDataExists(
+        `#${root}propertyAddress_propertyAddress1`,
+        firstLoan.propertyAddress.propertyAddress1,
+      );
+      cy.typeInIfDataExists(
+        `#${root}propertyAddress_propertyAddress2`,
+        firstLoan.propertyAddress.propertyAddress2,
+      );
+      cy.typeInIfDataExists(
+        `#${root}propertyAddress_propertyCity`,
+        firstLoan.propertyAddress.propertyCity,
+      );
+      cy.tabToElement(`#${root}propertyAddress_propertyState`);
+      cy.chooseSelectOptionUsingValue(firstLoan.propertyAddress.propertyState);
+      cy.typeInIfDataExists(
+        `#${root}propertyAddress_propertyZip`,
+        firstLoan.propertyAddress.propertyZip,
+      );
+
+      cy.typeInIfDataExists(`#${root}vaLoanNumber`, firstLoan.vaLoanNumber);
+      cy.tabToElement(`[name="${root}propertyOwned"]`);
+      cy.chooseRadio(firstLoan.propertyOwned ? 'Y' : 'N');
+      cy.tabToElement(`[name="${root}willRefinance"]`);
+      cy.chooseRadio(firstLoan.willRefinance ? 'Y' : 'N');
+    }
+
+    cy.tabToContinueForm();
+
+    // Upload supporting docs
+    // TODO: Figure out how to test keyboard only uploads
```

Comment:
Is this ticketed separately?

Reply:
I don't know if it's possible... when you select a file for upload, the browser opens a popup where you select the file and open it. [Cypress does have a command to do automatically](https://www.cypress.io/blog/2022/01/19/uploading-files-with-selectfile/), but we'd need to find a way to do with by keyboard only.

Reply:
Ah, I see. Do you think it would be sufficient to test only that the user can tab to the button that opens the popup? If the user is able to navigate to that button, then I think we could safely assume that their browser/os would allow them to choose the file they want, without actually testing it. What do you think?

Reply:
Hmm, maybe. I'm not sure if we can check that the popup has opened, but we should be able to close it using escape.

Reply:
Oh I'm actually thinking even simpler than that. Like, just checking that the user has the ability to tab to the button that would open the popup, if clicked. In other words, that `cy.tabToElement` doesn't raise an error if called with the selector for that button.